### DISCUSSION TITLE
HBITMAP to Bitmap in WinUI 3

### DISCUSSION THREAD

**Discussion by gautambjain** [microsoft/microsoft-ui-xaml#7921] [Q&A] [answered]
There using to be System.Drawing.Image.FromHbitmap in .Net Framework, but I don't know how to do the same in WinUI.

I have handle to HBITMAP. Have looked at several places. But don't know how to create objects of type Bitmap, BitmapSource, ImageSource from HBITMAP.

Basically I want to draw it on a control such as StackPanel etc.

Kindly please help.

**Comment by castorix** [comment, marked as answer]
I generally use a [WriteableBitmap](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.media.imaging.writeablebitmap?view=windows-app-sdk-1.2) (you can see for example the _GetCaptureWriteableBitmap_ function  that I used in [CMediaEngine.cs](https://github.com/castorix/WinUI3_MediaEngine/blob/master/CMediaEngine.cs) to convert a HBITMAP (_m_hBitmapCapture_) that I set as Image.Source in _SaveFrameDialog_ function in [MainWindow.xaml.cs](https://github.com/castorix/WinUI3_MediaEngine/blob/master/MainWindow.xaml.cs))

**Comment by gautambjain** [comment]
Thanks a lot @castorix . You have helped me several times . God bless. Today. on another forum also, I f[ound your answer](https://social.msdn.microsoft.com/Forums/vstudio/en-US/11bcd46b-ee7d-46f7-bcce-52b0a0bba453/looking-for-net-library-to-read-the-thumbnail-image-from-jpeg-file) very useful.

**Comment by peter0302** [comment]
So you're creating a new in-memory bitmap, rendering to it, and then copying it to the `WriteableBitmap`? Wouldn't it be much more memory efficient to render directly to the `WriteableBitmap`? Except it offers no way in C# to get the buffer, for... reasons.
